spec/requests/api/v3/courses/index_spec.rb

```diff
@@ -155,6 +155,57 @@
         expect(course_hashes.first["id"]).to eq(course_b.id.to_s)
         expect(course_hashes.second["id"]).to eq(course_a.id.to_s)
       end
+
+      context "when a course has 2 available sites" do
+        let(:another_site) { build(:site, latitude: 0.5, longitude: 0.5) }
+
+        before do
+          course_b.site_statuses << create(:site_status, :findable, site: another_site)
+        end
+
+        around do |example|
+          default_per_page = Kaminari.config.default_per_page
+
+          Kaminari.configure do |config|
+            config.default_per_page = 1
+          end
+
+          example.run
+
+          Kaminari.configure do |config|
+            config.default_per_page = default_per_page
+          end
+        end
+
+        let(:page_1) { "/api/v3/courses?sort=distance&latitude=0&longitude=0&page[page]=1" }
+        let(:page_2) { "/api/v3/courses?sort=distance&latitude=0&longitude=0&page[page]=2" }
+        let(:page_3) { "/api/v3/courses?sort=distance&latitude=0&longitude=0&page[page]=3" }
+
+        # jsonapi removes duplicates on the same page
+        # therefore this test needs to span multiple pages
+        # in order for the test to be valid
+
+        it "does not return duplicates" do
+          course_ids = []
+
+          get page_1
+
+          json_response = JSON.parse(response.body)
+          course_ids << json_response.dig("data", 0, "id")
+
+          get page_2
+
+          json_response = JSON.parse(response.body)
+          course_ids << json_response.dig("data", 0, "id")
+
+          get page_3
+
+          json_response = JSON.parse(response.body)
+          course_ids << json_response.dig("data", 0, "id")
+
+          expect(course_ids.compact.size).to eql(2)
```

Comment:
why `compact`?

Reply:
when the bug was present there were three pages. so the array had three items. it should only return 2 items. when the bug was fixed it returned `[1,2,nil]` as there is now nothing on the 3rd page. we have to compact to remove the extra result